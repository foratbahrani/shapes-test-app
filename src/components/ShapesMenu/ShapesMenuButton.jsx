import React from 'react';
import './ShapesMenuButton.scss';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectFilteredShapes } from '../../store/shapes/shapes.selectors';
import { toggleFilteredShape } from '../../store/shapes/shapes.actions';
import Shape from '../Shape/Shape';

export const ShapesMenuButton = ({ title, filteredShapes, toggleFilteredShape }) => {
	const isActive = filteredShapes.includes(title) ? 'shapesMenuButton--active' : '';
	return (
		<button
			className={`shapesMenuButton ${isActive} shrink`}
			id='button'
			onClick={() => toggleFilteredShape(title)}>
			<span className='lg'>{title}</span>
			<span className='md'>
				<Shape id='shape' size='sm' shape={title} color={isActive ? 'black' : ''} />
			</span>
		</button>
	);
};

const mapStateToProps = createStructuredSelector({
	filteredShapes: selectFilteredShapes,
});

const mapDispatchToProps = {
	toggleFilteredShape: toggleFilteredShape,
};

export default connect(mapStateToProps, mapDispatchToProps)(ShapesMenuButton);
