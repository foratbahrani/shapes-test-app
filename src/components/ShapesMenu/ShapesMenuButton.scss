.shapesMenuButton {
	appearance: none;
	border: none;
	background-color: transparent;
	color: inherit;
	font-weight: 600;
	font-size: 1.5em;
	font-style: italic;
	text-transform: uppercase;
	cursor: pointer;
	padding: 0;
	margin: 0 12px;
}

.shapesMenuButton:focus {
	outline: 0;
}

.shapesMenuButton--active {
	color: black;
}
