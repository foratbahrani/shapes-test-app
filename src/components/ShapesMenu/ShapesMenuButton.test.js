import { shallow } from 'enzyme';
import React from 'react';
import { ShapesMenuButton } from './ShapesMenuButton';

let wrapper;
let toggleFilteredShape = undefined;

beforeEach(() => {
	const shape = 'rectangle';
	const filteredShapes = ['rectangle'];
	toggleFilteredShape = jest.fn().mockReturnValue('');
	wrapper = shallow(
		<ShapesMenuButton
			color={shape}
			filteredShapes={filteredShapes}
			toggleFilteredShape={toggleFilteredShape}
		/>
	);
});

it('matches snapshot', () => {
	expect(wrapper).toMatchSnapshot();
});

it('calls toggle function', () => {
	expect.assertions(2);
	wrapper.find('[id="button"]').simulate('click');
	expect(toggleFilteredShape.mock.calls.length).toBe(1);
	wrapper.find('[id="button"]').simulate('click');
	expect(toggleFilteredShape.mock.calls.length).toBe(2);
});

it('activates when needed', () => {
	expect(
		shallow(<ShapesMenuButton filteredShapes={['rectangle']} title='rectangle' />)
			.find("[id='button']")
			.props().className
	).toEqual('shapesMenuButton shapesMenuButton--active shrink');
});

it('deactivates when needed', () => {
	expect(
		shallow(<ShapesMenuButton filteredShapes={['square']} title='rectangle' />)
			.find("[id='button']")
			.props().className
	).toEqual('shapesMenuButton  shrink');
});
