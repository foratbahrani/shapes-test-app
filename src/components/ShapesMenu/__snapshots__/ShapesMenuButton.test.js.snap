// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="shapesMenuButton  shrink"
  id="button"
  onClick={[Function]}
>
  <span
    className="lg"
  />
  <span
    className="md"
  >
    <Shape
      color=""
      id="shape"
      size="sm"
    />
  </span>
</button>
`;
